packages/circuit-ui/components/Popover/Popover.spec.tsx

```diff
@@ -99,21 +101,26 @@ describe('PopoverItem', () => {
 });
 
 describe('Popover', () => {
-  const Default = (props: Omit<PopoverProps, 'triggerRef'>) => {
-    const triggerRef = useRef<HTMLButtonElement & HTMLAnchorElement>(null);
-
-    return (
-      <Fragment>
-        <button ref={triggerRef}>Button</button>
-        <Popover triggerRef={triggerRef} {...props} />
-      </Fragment>
+  const renderPopover = (props: Omit<PopoverProps, 'component'>) =>
+    render(
+      <Popover
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        component={forwardRef<HTMLElement>((props, ref) => {
+          const buttonRef = ref as Ref<HTMLButtonElement>;
```

Comment:
@connor-baer any way to avoid the typecast here or to do it in a nicer way? This is the only way I've found to make TS happy so far 😕 

The Popover expects a `component` with a `Ref<HTMLElement>` because it won't necessarily be a button, it could also be the `ImageInput`, but the forwarded `Ref<HTMLElement>` clashes with the `Ref<HTMLButtonElement>` that a buttonexpects.

Reply:
The Popover should expect a `Ref<HTMLButtonElement>`. Even inside the `ImageInput`, the trigger component should be a button, right? 

If that's too strict, I'd use `Ref<any>` in the Popover instead.

Reply:
Hm,  that's true, it. should probably be a `HTMLButtonElement` even for the `ImageInput` 👀  We can probably try this for now and it wouldn't be a breaking change if we had to make it more flexible in the future (or would it?)

Would it still clash when we use a Circuit `Button` though? Since it's a `Ref<HTMLButtonElement & HTMLAnchorElement>`?

Reply:
Sounds good 👍🏻 Loosening restriction would not be a breaking change. 

> Would it still clash when we use a Circuit `Button` though? Since it's a `Ref<HTMLButtonElement & HTMLAnchorElement>`?

I'm not sure, it might.

Reply:
You could also change this line: 
```diff
-        component={forwardRef<HTMLElement>((props, ref) => {
+        component={forwardRef<HTMLButtonElement>((props, ref) => {
```

TypeScript should be okay with that.